[config/configtls] build-and-test failing on arm

### Component(s)

[config/configtls]

### What happened?

**Describe the bug**
recent PR #12801  to [config/configtls] introduced a dependency on OpenSSL/CGO. This is causing the unit tests to fail on MacOS, both locally or in pipeline

**Steps to reproduce**
run `make gotest` on MacOS ARM or click to see most recent pipeline CI status: https://github.com/open-telemetry/opentelemetry-collector/actions/runs/14841117409/job/41664078361

**What did you expect to see?**
tests passing or skipped

**What did you see instead?**
tests failing

```
# github.com/google/go-tpm-tools/simulator/internal
In file included from /Users/runner/go/pkg/mod/github.com/google/go-tpm-tools@v0.4.4/simulator/internal/internal_cgo.go:45:
In file included from /Users/runner/go/pkg/mod/github.com/google/go-tpm-tools@v0.4.4/simulator/ms-tpm-20-ref/TPMCmd/tpm/include/Tpm.h:47:
In file included from /Users/runner/go/pkg/mod/github.com/google/go-tpm-tools@v0.4.4/simulator/ms-tpm-20-ref/TPMCmd/tpm/include/LibSupport.h:65:
/Users/runner/go/pkg/mod/github.com/google/go-tpm-tools@v0.4.4/simulator/ms-tpm-20-ref/TPMCmd/tpm/include/Ossl/TpmToOsslSym.h:47:10: fatal error: 'openssl/aes.h' file not found
#include <openssl/aes.h>
         ^~~~~~~~~~~~~~~
1 error generated.
✖  .
make[2]: *** [test] Error 1

make[1]: *** [config/configtls] Error 2
=== Failed
=== FAIL: .  (0.00s)
make[1]: *** Waiting for unfinished jobs....
FAIL	go.opentelemetry.io/collector/config/configtls [build failed]

=== Errors
In file included from /Users/runner/go/pkg/mod/github.com/google/go-tpm-tools@v0.4.4/simulator/internal/internal_cgo.go:45:
In file included from /Users/runner/go/pkg/mod/github.com/google/go-tpm-tools@v0.4.4/simulator/ms-tpm-20-ref/TPMCmd/tpm/include/Tpm.h:47:
In file included from /Users/runner/go/pkg/mod/github.com/google/go-tpm-tools@v0.4.4/simulator/ms-tpm-20-ref/TPMCmd/tpm/include/LibSupport.h:65:
/Users/runner/go/pkg/mod/github.com/google/go-tpm-tools@v0.4.4/simulator/ms-tpm-20-ref/TPMCmd/tpm/include/Ossl/TpmToOsslSym.h:47:10: fatal error: 'openssl/aes.h' file not found
#include <openssl/aes.h>
         ^~~~~~~~~~~~~~~
1 error generated.
```

### Collector version

v0.125.0

## Environment
OS: macOS 14/15
Platform: ARM

--- a/.chloggen/tpm-test-darwin.yaml
+++ b/.chloggen/tpm-test-darwin.yaml
@@ -1,0 +1,25 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: bug_fix
+
+# The name of the component, or a single word describing the area of concern, (e.g. otlpreceiver)
+component: tlsconfig
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: "Disable TPM tests on MacOS/Darwin"
+
+# One or more tracking issues or pull requests related to the change
+issues: [12964]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext:
+
+# Optional: The change log or logs in which this entry should be included.
+# e.g. '[user]' or '[user, api]'
+# Include 'user' if the change is relevant to end users.
+# Include 'api' if there is a change to a library API.
+# Default: '[user]'
+change_logs: []

--- a/config/configtls/tpm_test.go
+++ b/config/configtls/tpm_test.go
@@ -2,7 +2,7 @@
 // SPDX-License-Identifier: Apache-2.0
 
 // Don't run this test on Windows, as it requires a TPM simulator which depends on openssl headers.
-//go:build !windows
+//go:build !windows && !darwin
 
 package configtls // import "go.opentelemetry.io/collector/config/configtls"
 

